File: 프로그래머스/2018 KAKAO BLIND RECRUITMENT/[3차]n진수 게임.py
# [3차]n진수 게임
def convert(total,base):
    char = ['0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F']
    result = '0'
    num = 0
    while len(result) <= total:
        cur = num
        tmp = ''
        while cur > 0:
            tmp += char[cur % base]
            cur //= base
        result += tmp[::-1]
        num += 1
    return result

def solution(n, t, m, p):
    answer = ''
    numbers = convert(m*t,n)
    for i in range(t):
        answer += numbers[(i*m)+p-1]
    return answer 

File: 프로그래머스/2018 KAKAO BLIND RECRUITMENT/[3차]n진수 게임.js
// [3차]n진수 게임
function convert(total, base){
    const change = '0123456789ABCDEF';
    let result = '0';
    let num = 0;
    while(result.length <= total){
        let cur = num;
        let tmp = '';
        while (cur > 0){
            tmp += change[cur % base];
            cur = parseInt(cur/base);
        }
        result += tmp.split("").reverse().join("");
        num += 1;
    }
    return result
}
function solution(n, t, m, p) {
    const result = convert(t*m,n).split("");
    console.log(result)
    if(p == m) p = 0
    const answer = result.filter((e,i) => (i+1) % m == p && i / m < t).join("");
    return answer;
}